From a5b3dd887dcd60e981626ae2e15c874311fc4fef Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Sun, 2 Sep 2012 04:36:13 +0200
Subject: [PATCH] Send resume signal when built with systemd and using
 pm-utils suspend

This case was apparently overlooked when adding systemd support.

Signed-off-by: Jan Alexander Steffens (heftig) <jan.steffens@gmail.com>
---
 src/linux/up-backend.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/linux/up-backend.c b/src/linux/up-backend.c
index ddd4472..52b430e 100644
--- a/src/linux/up-backend.c
+++ b/src/linux/up-backend.c
@@ -603,10 +603,11 @@ gboolean
 up_backend_emits_resuming (UpBackend *backend)
 {
 #ifdef HAVE_SYSTEMD
-	return TRUE;
-#else
-	return FALSE;
+	if (sd_booted ())
+		return TRUE;
+	else
 #endif
+	return FALSE;
 }

 /**
--
1.7.12
